packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -2142,6 +2169,17 @@ export const showNonGenuineMongoDBWarningModal = (
   };
 };
 
+export const showEndOfLifeMongoDBWarningModal = (
+  connectionId: string,
+  version: string
+): ConnectionsThunkAction<void> => {
+  return (_dispatch, getState, { track }) => {
+    const connectionInfo = getCurrentConnectionInfo(getState(), connectionId);
+    track('Screen', { name: 'end_of_life_mongodb_modal' }, connectionInfo);
+    void _showEndOfLifeMongoDBWarningModal(connectionInfo, version);
```

Comment:
This is a nit, but I'd personally probably have jumped through hoops to come up with two different names where one showEndOfLifeMongoDBWarningModal is the thunk action creator thingy and one is the function from the component. But I can live with this ;)

Reply:
TBH, I didn't pay too much attention to naming there as I was mostly copying from the code above:

https://github.com/mongodb-js/compass/blob/32e0d30fbe6943c05f4143286831bf278eb900ce/packages/compass-connections/src/stores/connections-store-redux.ts#L2162-L2171